Mike Hulbert/week_01/arrays/js/exercise.js

```diff
@@ -0,0 +1,117 @@
+// 2. Define a function `maxOfThree` that takes three numbers as arguments and returns the largest of them.
+// 3. Write a function that takes a character (i.e. a string of length 1) and returns true if it is a vowel, false otherwise.
+// 4. Define a function `sumArray` and a function `multiplyArray` that sums and multiplies (respectively) all the numbers in an array of numbers. For example, `sumArray([1,2,3,4])` should return 10, and `multiplyArray([1,2,3,4])` should return 24.
+//
+//----------------------------------------Max of Two----------------------------
+
+const maxOfTwoNumbers = function (a,b) {
+  if (a > b) {
+    return a;
+  } else {
+    return b;
+  }
+};
+console.log(maxOfTwoNumbers(20,3));
+
+//----------------------------------------Max of Three--------------------------
+
+const maxOfThree = function (a, b, c) {
+  if (a > b && a > c) {
+    return a;
+  } else if (b > a && b > c) {
+    return b;
+  } else {
+    return c;
+  }
+};
+console.log(maxOfThree(11,12,13));
+
+//----------------------------------------True or False-------------------------
+
+const word = "abcdefghijklmnopqrstuvwxyz";
+
+for (let i = 0; i < word.length; i++){
+  word[i];
+  if (word[i] === "a" || word[i] === "e" || word[i] === "i" || word[i] === "o" || word[i] === "u"){
+  console.log(word[i],true);
+  } else {
+  console.log(word[i],false);
+  }
+}
+
+//---------------------------------------Sum & Multiply Arrays------------------
+
+const numbers = [1, 2, 3, 4]
+
+const sumArray = function (a) { // sum of numbers
+  for (let i = 0; i < a.length; i++) {
+    console.log(`${ a[0] + a[1] + a[2] + a[3] }`);
+  }
+}
+sumArray(numbers);
+
+const multiplyArray = function (a) { // multiply numbers
+  for (let i = 0; i < a.length; i++) {
+    console.log(`${ a[0] * a[1] * a[2] * a[3] }`);
+  }
+}
+multiplyArray(numbers);
+
+// ## Bonus
+//
+// 5. Define a function `reverseString` that computes the reversal of a string. For example, reverseString("jag testar") should return the string "ratset gaj".
+// 6. Write a function `findLongestWord` that takes an array of words and returns the length of the longest one.
+// 7. Write a function `filterLongWords` that takes an array of words and an number `i` and returns the array of words that are longer than i.
+
+// # Homework: The Word Guesser
+
+//------------------------------------------Reverse String----------------------
+
+const reverseString = function (str) {
+  return str.split("").reverse().join("");
+}
+console.log(reverseString("jag testar"));
+
+//-----------------------------------------Find Longest word--------------------
+
+let result = '';
+const words = ['none', 'two', 'three', 'four', 'onehundred'];
+
+const findLongestWord = function (array) {
+  for (let i = 0; i < array.length; i++) {
+    str = array[i].split('').join('');
+    if (str.length >= 10) { // ??  compare results from loop?
+      result = str;
+      return result ;
+    }
+  }
+};
```

Comment:
Here's some pseudo code for this function that might help. Feel free to give it another go and re-submit it. 

const findLongestWord = function (array) {
    // initialise the longest variable to be 0
    // loop through the array
        // if the length of the word is longer than the longest variable
            // reassign the longest variable to be the length of the word
        // 
    // end of the loop
    // return the longest variable 
}